cmd/main.go

```diff
@@ -241,475 +340,84 @@ Stacks may include:
-		return BlockMeta(*blockMetaBlockID)
-
-	case blockIgnoreCmd.FullCommand():
-		return BlockIgnore(*blockIgnoreBlockID)
-
-	case blockFileCmd.FullCommand():
-		return BlockFile(*blockFileBlockID, *blockFileIndex, *blockFilePath, *blockFileContent)
-
-	// cafe
-	case cafeAddCmd.FullCommand():
-		return CafeAdd(*cafeAddPeerID, *cafeAddToken)
-
-	case cafeListCmd.FullCommand():
-		return CafeList()
-
-	case cafeGetCmd.FullCommand():
-		return CafeGet(*cafeGetCafeID)
-
-	case cafeDeleteCmd.FullCommand():
-		return CafeDelete(*cafeDeleteCafeID)
-
-	case cafeMessagesCmd.FullCommand():
-		return CafeMessages()
-
-	// chat
-	case chatCmd.FullCommand():
-		return Chat(*chatThreadID)
-
-	// comments
-	case commentAddCmd.FullCommand():
-		return CommentAdd(*commentAddBlockID, *commentAddBody)
-
-	case commentListCmd.FullCommand():
-		return CommentList(*commentListBlockID)
-
-	case commentGetCmd.FullCommand():
-		return CommentGet(*commentGetBlockID)
-
-	case commentIgnoreCmd.FullCommand():
-		return CommentIgnore(*commentIgnoreBlockID)
-
-	// config
-	case configCmd.FullCommand():
-		return Config(*configName, *configValue)
-
-	// contacts
-	case contactAddCmd.FullCommand():
-		return ContactAdd(*contactAddName, *contactAddAddress, *contactAddWait)
-
-	case contactListCmd.FullCommand():
-		return ContactList()
-
-	case contactGetCmd.FullCommand():
-		return ContactGet(*contactGetAddress)
-
-	case contactDeleteCmd.FullCommand():
-		return ContactDelete(*contactDeleteAddress)
+	// file list
+	fileListCmd := fileCmd.Command("list", `Get all the files, or just the files for a specific thread or block`).Alias("ls").Default()
 
-	case contactSearchCmd.FullCommand():
-		return ContactSearch(*contactSearchName, *contactSearchAddress, *contactSearchLocal, *contactSearchRemote, *contactSearchLimit, *contactSearchWait)
+	// file list thread
+	fileListThreadCmd, fileListThreadCallback := threadFilesCommand(fileListCmd, []string{"thread"})
+	cmds[fileListThreadCmd.Default().FullCommand()] = fileListThreadCallback
 
-	// daemon
-	case daemonCmd.FullCommand():
-		repoPath, err := getRepoPath(*daemonRepoPath)
-		if err != nil {
-			return err
-		}
-		return Daemon(repoPath, *daemonPinCode, *daemonDocs, *logDebug)
+	// file list block
+	fileListBlockCmd, fileListBlockCallback := blockFilesCommand(fileListCmd, []string{"block"})
+	cmds[fileListBlockCmd.FullCommand()] = fileListBlockCallback
 
-	// docs
-	case docsCmd.FullCommand():
-		return Docs()
-
-	// feed
-	case feedCmd.FullCommand():
-		return Feed(*feedThreadID, *feedOffset, *feedLimit, *feedMode)
-
-	// file
-	case fileListCmd.FullCommand():
-		return FileList(*fileListThreadID, *fileListOffset, *fileListLimit)
-
-	case fileKeysCmd.FullCommand():
+	// file keys
+	fileKeysCmd      := fileCmd.Command("keys", "Shows file keys under the given target").Alias("key")
+	fileKeysTargetID := fileKeysCmd.Arg("target-block", "Files Block Target ID").Required().String()
+	// @todo why is this a block target
```

Comment:
Because there is a different key for each content/meta pair in the DAG that target points to

Reply:
(so keys lists key to each pair by referencing their paths)